docs/design/quota.md

```diff
@@ -0,0 +1,121 @@
+# NooBaa support for bucket quota
+
+## Overview
+Quota is a known tool for limiting the max total size / max number of objects per bucket.
+
+By having this limitation on noobaa object buckets, we can avoid resource starvation by a single consumer.
+
+## Goals
+* Provide an option to Configure Quota per Bucket on an OBC level and on the BucketClass level. If both exist, the OBC definition would override the BucketClass definition. 
+* Set a quota of object bucket claim, which limits the max number of objects per bucket (MaxObjects).
+* Set a quota of object bucket claim, which limits the max total size per bucket (MaxSize).
+* Remove quota configuration from the bucket. **?**
+* Update Quota’s on exiting buckets (updating the OBC only).
+* Update BucketClass quota, will be effective in new buckets using this BucketClass.
+* Add to NooBaa CLI new options for creating object bucket claims while providing one of the 2 types of quota.
+* Metrics and Alerts - upon calculation statistics based:
+    * Will provide the admin the quota status.
+    * Alert the admin when the number of objects / total size of objects reaches 80% and 100% of the quota. **# in the code 90% and 100% threashold?**
```

Comment:
@nimrod-becker what should be the right threshold? should we change the existing behavior to alert on 80%?

Reply:
Sure, lets do 80 and 100.
In any case, I think the more important alerts are the mixings

Reply:
I suggest to pull the threshold's values out to config file (in json format) or env. then we can easy manage it.  
What do you think?